Plugin: ChatGPT官方接口版

**插件名称：**

ChatGPT官方接口版

**插件功能：**

适配新版OpenAI API的nonebot插件，可以调用官方稳定版ChatGPT进行回复

**PyPI 项目名：**

nonebot-plugin-chatgpt-turbo

**插件 import 包名：**

nonebot_plugin_chatgpt_turbo

**插件项目仓库/主页链接：**

https://github.com/Alpaca4610/nonebot_plugin_chatgpt_turbo

**标签：**

[{"label":"ChatGPT","color":"#ea5252"},{"label":"OpenAI","color":"#52ea92"}]

# 📃 商店发布检查结果

> Plugin: ChatGPT官方接口版

**⚠️ 在发布检查过程中，我们发现以下问题：**
<pre><code><li>⚠️ 插件加载测试未通过。<details><summary>测试输出</summary>插件 nonebot-plugin-chatgpt-turbo 的信息如下：
    name         : nonebot-plugin-chatgpt-turbo             
     version      : 0.1.0                                    
     description  : A nonebot plugin for openai gpt3.5-turbo 
    
    dependencies
     - httpx &gt;=0.23.3,&lt;0.24.0
     - nonebot-adapter-onebot &gt;=2.2.1,&lt;3.0.0
     - nonebot2 &gt;=2.0.0rc3,&lt;3.0.0
     - openai &gt;=0.27.0,&lt;0.28.0
插件 nonebot_plugin_chatgpt_turbo 加载出错：
03-02 17:20:32 [SUCCESS] nonebot | NoneBot is initializing...
03-02 17:20:32 [INFO] nonebot | Current Env: prod
03-02 17:20:32 [ERROR] nonebot | Failed to import &quot;nonebot_plugin_chatgpt_turbo&quot;
Traceback (most recent call last):
  File &quot;/home/runner/work/nonebot2/nonebot2/plugin_test/runner.py&quot;, line 4, in &lt;module&gt;
    valid = load_plugin(&quot;nonebot_plugin_chatgpt_turbo&quot;)
  File &quot;/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/load.py&quot;, line 34, in load_plugin
    return manager.load_plugin(module_path)
&gt; File &quot;/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/manager.py&quot;, line 141, in load_plugin
    module = importlib.import_module(name)
  File &quot;/usr/lib/python3.10/importlib/__init__.py&quot;, line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1050, in _gcd_import
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1027, in _find_and_load
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 1006, in _find_and_load_unlocked
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 688, in _load_unlocked
  File &quot;/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/manager.py&quot;, line 233, in exec_module
    super().exec_module(module)
  File &quot;&lt;frozen importlib._bootstrap_external&gt;&quot;, line 883, in exec_module
  File &quot;&lt;frozen importlib._bootstrap&gt;&quot;, line 241, in _call_with_frames_removed
  File &quot;/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot_plugin_chatgpt_turbo/__init__.py&quot;, line 10, in &lt;module&gt;
    api_key = nonebot.get_driver().config.openai_api_key
AttributeError: &#x27;Config&#x27; object has no attribute &#x27;openai_api_key&#x27;</details></li></code></pre>
<details><summary>详情</summary><pre><code><li>✅ 标签: ChatGPT-#ea5252, OpenAI-#52ea92。</li><li>✅ 项目 <a href="https://github.com/Alpaca4610/nonebot_plugin_chatgpt_turbo">主页</a> 返回状态码 200。</li><li>✅ 包 <a href="https://pypi.org/project/nonebot-plugin-chatgpt-turbo/">nonebot-plugin-chatgpt-turbo</a> 已发布至 PyPI。</li></code></pre></details>

---

💡 如需修改信息，请直接修改 issue，机器人会自动更新检查结果。
💡 当插件加载测试失败时，请发布新版本后在当前页面下评论任意内容以触发测试。

💪 Powered by [NoneBot2 Publish Bot](https://github.com/nonebot/nonebot2-publish-bot)
<!-- PUBLISH_BOT -->

> # 📃 商店发布检查结果
> > Plugin: ChatGPT官方接口版
> 
> **⚠️ 在发布检查过程中，我们发现以下问题：**
> 
> ```
> * ⚠️ 插件加载测试未通过。测试输出插件 nonebot-plugin-chatgpt-turbo 的信息如下：
>       name         : nonebot-plugin-chatgpt-turbo             
>        version      : 0.1.0                                    
>        description  : A nonebot plugin for openai gpt3.5-turbo 
>       
>       dependencies
>        - httpx >=0.23.3,<0.24.0
>        - nonebot-adapter-onebot >=2.2.1,<3.0.0
>        - nonebot2 >=2.0.0rc3,<3.0.0
>        - openai >=0.27.0,<0.28.0
>   插件 nonebot_plugin_chatgpt_turbo 加载出错：
>   03-02 17:33:35 [SUCCESS] nonebot | NoneBot is initializing...
>   03-02 17:33:35 [INFO] nonebot | Current Env: prod
>   03-02 17:33:36 [ERROR] nonebot | Failed to import "nonebot_plugin_chatgpt_turbo"
>   Traceback (most recent call last):
>     File "/home/runner/work/nonebot2/nonebot2/plugin_test/runner.py", line 4, in <module>
>       valid = load_plugin("nonebot_plugin_chatgpt_turbo")
>     File "/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/load.py", line 34, in load_plugin
>       return manager.load_plugin(module_path)
>   > File "/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/manager.py", line 141, in load_plugin
>       module = importlib.import_module(name)
>     File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
>       return _bootstrap._gcd_import(name[level:], package, level)
>     File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>     File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>     File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
>     File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
>     File "/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot/plugin/manager.py", line 233, in exec_module
>       super().exec_module(module)
>     File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>     File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>     File "/home/runner/.cache/pypoetry/virtualenvs/plugin-test-rYmcgvOP-py3.10/lib/python3.10/site-packages/nonebot_plugin_chatgpt_turbo/__init__.py", line 10, in <module>
>       api_key = nonebot.get_driver().config.openai_api_key
>   AttributeError: 'Config' object has no attribute 'openai_api_key'
> ```
> 
> 详情
> ```
> * ✅ 标签: ChatGPT-#ea5252, OpenAI-#52ea92。* ✅ 项目 [主页](https://github.com/Alpaca4610/nonebot_plugin_chatgpt_turbo) 返回状态码 200。* ✅ 包 [nonebot-plugin-chatgpt-turbo](https://pypi.org/project/nonebot-plugin-chatgpt-turbo/) 已发布至 PyPI。
> ```
> 
> 💡 如需修改信息，请直接修改 issue，机器人会自动更新检查结果。 💡 当插件加载测试失败时，请发布新版本后在当前页面下评论任意内容以触发测试。
> 
> ♻️ 评论已更新至最新检查结果
> 
> 💪 Powered by [NoneBot2 Publish Bot](https://github.com/nonebot/nonebot2-publish-bot)

test

--- a/website/static/plugins.json
+++ b/website/static/plugins.json
@@ -4665,5 +4665,24 @@
       }
     ],
     "is_official": false
+  },
+  {
+    "module_name": "nonebot_plugin_chatgpt_turbo",
+    "project_link": "nonebot-plugin-chatgpt-turbo",
+    "name": "ChatGPT官方接口版",
+    "desc": "适配新版OpenAI API的nonebot插件，可以调用官方稳定版ChatGPT进行回复",
+    "author": "Alpaca4610",
+    "homepage": "https://github.com/Alpaca4610/nonebot_plugin_chatgpt_turbo",
+    "tags": [
+      {
+        "label": "ChatGPT",
+        "color": "#ea5252"
+      },
+      {
+        "label": "OpenAI",
+        "color": "#52ea92"
+      }
+    ],
+    "is_official": false
   }
 ]

